server/tests/server/Group.test.js

```diff
@@ -0,0 +1,178 @@
+import chai from 'chai';
+import supertest from 'supertest';
+import app from '../../config/app';
+import db from '../../models';
+import groupHelper from '../Helpers/GroupHelper';
+import userHelper from '../Helpers/UserHelper';
+import messageHelper from '../Helpers/MessageHelper';
+
+
+const expect = chai.expect;
+const request = supertest(app);
+const goodMessage = messageHelper.goodMessage;
+const badMessage = messageHelper.badMessage;
+const goodGroup = groupHelper.goodGroup1;
+const anotherGroup = groupHelper.goodGroup2;
+const user = userHelper.anotherUser;
+const userDoesntExist = userHelper.goodUser;
+const badGroup = groupHelper.badGroup;
+const username = userHelper.userName;
+let groupId;
+let userResponse;
+
+
+describe('POST api/group', () => {
+  beforeEach((done) => {
+    request.post('/api/v1/user/signup')
+    .send(user)
+    .end((err, res) => {
+      if (err) return err;
+      userResponse = res.body;
+      request.post('/api/v1/group')
+      .set({ Authorization: userResponse.token })
+      .send(goodGroup)
+      .end((err, res) => {
+        if (err) return err;
+        done();
+      });
+    });
+  });
+  after(() => db.sequelize.sync({ force: true }));
+  it('should create a new group', (done) => {
+    request.post('/api/v1/group')
+    .send(anotherGroup)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      if (err) return err;
+      expect(res.status).to.equal(200);
+      expect(res.body).to.have.property('name');
+      expect(res.body).to.have.property('description');
+      done();
+    });
+  });
+  it('should not create group with missing property ', (done) => {
+    request.post('/api/v1/group')
+    .send(badGroup)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      expect(res.body.message).to.equal('unexpected error occured');
+      return done();
+    });
+  });
+  it('should not create a group if already exist ', (done) => {
+    request.post('/api/v1/group')
+    .send(goodGroup)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      expect(res.status).to.equal(409);
+      expect(res.body.message).to.equal('Group already exist');
+      return done();
+    });
+  });
+  it('should add users to group', (done) => {
+    request.post('/api/v1/group/1/user')
+    .send(username)
+    .set({ Authorization: userResponse.token })
+    .end((err, res) => {
+      expect(res.status).to.equal(201);
```

Comment:
When writing test, it's imperative that what you intend to test for matches what's actually being tested for. It's also important that your test descriptions match what's actually being tested for. Please go through the other tests suites in your code base and refactor appropriately.